pkg/controllers/deprovisioning/controller.go

```diff
@@ -115,32 +116,13 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// Attempt different deprovisioning methods. We'll only let one method perform an action
 	isConsolidated := c.cluster.Consolidated()
 	for _, d := range c.deprovisioners {
-		candidates, err := GetCandidates(ctx, c.cluster, c.kubeClient, c.clock, c.cloudProvider, d.ShouldDeprovision)
+		success, err := c.deprovision(ctx, d)
 		if err != nil {
-			return reconcile.Result{}, fmt.Errorf("determining candidates, %w", err)
+			logging.FromContext(ctx).Info("could not deprovision, %s", d, err)
```

Comment:
Can we aggregate the errors from each deprovisioner into a multierr and then log if they all fail? I fear that we'll get lots of "could not deprovision" lines for successful deprovisioning loops, which may be confusing. If we log all errors on a no-op or errored loop, we're interested in all deprovisioners that failed.

Reply:
Decided against this one. If one deprovisioner fails with an error (validation, etc) we want to be noisy about it. If all but one fail, we want to know. Realistically, 4/5 failing should only happen when something is seriously wrong.

Reply:
Even if we're not doing a multierr if all fail, I think for logging purposes, it would be less noisy if we did a multierr even on a success. I think it's fine either way, but any reasonable way we can reduce the number of log lines I think would help here for the visibility story.

Reply:
I played with it, but then we end up with something like:

"deprovisioning via delete, ..."
"failed to deprovision via expiration xyz, via consolidate, xyz"

I found this confusing.

Much more clear to have the failures come first and the success log come later. I also don't think we're going to be getting so many errors that this will impact customers regularly.

Reply:
There's potentially a better argument for erroring immediately, but making sure that deprovisioners can't error in a blocking way.